src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java

```diff
@@ -4033,6 +4033,10 @@ protected JCClassDecl enumDeclaration(JCModifiers mods, Comment dc) {
 
         Name name = typeName();
 
+        if(typeParametersOpt().size() > 0) {
+            log.error(DiagnosticFlag.SYNTAX, S.prevToken().endPos, Errors.EnumCantBeGeneric);
```

Comment:
I agree with Pavel the location of the error could be better - probably store the `typeParametersOpt` into a variable, and then use the content to provide a better location. Or, you could store some meaningful position based on tokens.

Reply:
I improved the error position. Now all the following report the same error:

```java
  public enum E1<> {}
  public enum E2<T> {}
  public enum E3<T, T> {}
```

However, at a cost of a small refactoring in `typeParametersOpt` to allow (erroneous) empty lists of type parameters opt. The new version (the overload with the boolean) can be used if the caller wants to return a more precise error as in this case. The way to communicate this situation to the caller is through a `null` ret list. Despite I noticed that `null` is used in various cases, I am still a bit skeptical if this is the right way (for such small benefit).

WDYT @pavelrappo, @lahodaj